File: srm-portal-web/src/api/contract.ts
import { api } from './http'

export type PortalContractSummary = {
  id: number
  contractNo: string
  title: string
  status: string
  contractType: string
  supplierId: number
  supplierCode: string
  supplierName: string
  procurementOrgId: number
  procurementOrgCode: string
  startDate: string | null
  endDate: string | null
  totalAmount: string | null
  currency: string
}

export type PortalContractLine = {
  id: number
  lineNo: number
  materialId: number | null
  materialCode: string | null
  materialName: string | null
  materialDesc: string | null
  qty: string | null
  uom: string | null
  unitPrice: string | null
  amount: string | null
  remark: string | null
}

export type PortalContractDetail = PortalContractSummary & {
  remark: string | null
  lines: PortalContractLine[]
}

export const portalContractApi = {
  list: () => api.get<PortalContractSummary[]>('/api/v1/portal/contracts'),
  get: (id: number) => api.get<PortalContractDetail>(`/api/v1/portal/contracts/${id}`),
}

File: srm-backend/src/main/java/com/srm/contract/domain/Contract.java
package com.srm.contract.domain;

import com.srm.foundation.domain.BaseEntity;
import com.srm.foundation.domain.OrgUnit;
import com.srm.master.domain.Supplier;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "contract")
public class Contract extends BaseEntity {

    @Column(name = "contract_no", nullable = false, unique = true, length = 64)
    private String contractNo;

    @Column(nullable = false)
    private String title;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "supplier_id", nullable = false)
    private Supplier supplier;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "procurement_org_id", nullable = false)
    private OrgUnit procurementOrg;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 32)
    private ContractStatus status = ContractStatus.DRAFT;

    @Column(name = "contract_type", length = 32)
    private String contractType = "FRAMEWORK";

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(name = "total_amount", precision = 19, scale = 4)
    private BigDecimal totalAmount;

    @Column(nullable = false, length = 8)
    private String currency = "CNY";

    @Column(length = 2000)
    private String remark;

    @OneToMany(mappedBy = "contract", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("lineNo ASC")
    private List<ContractLine> lines = new ArrayList<>();
}
